elasticsearch/run.sh

```diff
@@ -216,4 +216,8 @@ verify_or_add_index_templates() {
 
 verify_or_add_index_templates &
 
+if [ -v HEAP_DUMP_LOCATION ]; then
```

Comment:
What does `-v` do?  Are  you trying to test if the variable HEAP_DUMP_LOCATION is set to a non-empty value?  If so, use 
```
   if [ -n "${HEAP_DUMP_LOCATION:-}" ] ; then
```